src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/rmd/TextEditingTargetNotebook.java

```diff
@@ -1435,14 +1437,20 @@ public void execute()
    private void execAlternateEngineChunk(String docId,
                                          String chunkId,
                                          String engine,
-                                         String code)
+                                         String code,
+                                         Map<String, String> options)
    {
+      JsObject jsOptions = JsObject.createJsObject();
+      for (Map.Entry<String, String> entry : options.entrySet())
+         jsOptions.setString(entry.getKey(), entry.getValue());
```

Comment:
@jjallaire just want to confirm all is well with cross-window marshaling here (passing JavaScript strings across windows seems fine)